### DISCUSSION TITLE
How to do a global Sharepoint search?

### DISCUSSION THREAD

**Discussion by dicknetherlands** [O365/python-o365#958] [Q&A]
Via my browser I am able to perform a search across a whole sharepoint organisation that returns a list of hits with titles, summary contents (including highlighted search terms) and URLs to the matching resources. This searches everything including all documents, pages, files, lists, etc., across all sites in my organisation, all at once.

For example the following URL does this for all objects mentioning "guidelines" at all sharepoint sites in the sharepoint organisation for "myorganisation":

  https://myorganisation.sharepoint.com/search/Pages/results.aspx?k=guidelines

Is there any way to do this with python-o365? Or a way to approximate it?

**Comment by alejcas** [comment]
Not for the moment sorry

**Reply by dicknetherlands** [reply]
Thanks for the reply. Is there a way to use the python-o365 library to log in and provide authentication to that URL to call it directly instead?

**Reply by dicknetherlands** [reply]
Or, in other words, how do i use python-o365 to call an API endpoint that is not defined in the library itself, e.g. against the equivalent API URL to the above browser-based search? https://myorganisaiton.sharepoint.com/_api/search/query?querytext=%27guidelines%27

**Reply by alejcas** [reply]
You ca do this, but you will need to implement it yourself. 
To do so, you will need to read the library code to understand how it works.
